lib/AST/ASTPrinter.cpp

```diff
@@ -5050,6 +5098,13 @@ void PrintAST::visitErasureExpr(ErasureExpr *expr) {
 }
 
 void PrintAST::visitKeyPathExpr(KeyPathExpr *expr) {
+  auto components = expr->getComponents();
+  for (int i = 0; i <= (int) components.size() - 1; i++) {
```

Comment:
You can do `for (auto i : indices(components))` to iterate over the indices of a collection.

Reply:
Yeah, much better.

Reply:
But I used the normal iterative version to ensure the `'.'` in the `KeyPathExpr`.